tests/test_connection.py

```diff
@@ -48,6 +48,9 @@ def test_connect(node_factory):
     assert len(l1.rpc.listpeers()) == 1
     assert len(l2.rpc.listpeers()) == 1
 
+    if EXPERIMENTAL_FEATURES:
+        l1.daemon.wait_for_log("Peer reports remote_addr: 127.0.0.1:[0-9]{5}")
```

Comment:
You can actually check this is correct, with l1.port?

Reply:
Maybe also change message to say "Peer tells us it sees our our address as "? which is clearer for normal users...

Reply:
I can't check for l1 listening port, because we see the source port of the outbound connection which is dynamically assigned by the kernel. I added a question to the RFC about that: https://github.com/lightning/bolts/pull/917#issuecomment-1008276220

Reply:
Changed the wording though. Keep in mind that `remote_addr` is a protocol field.